### DISCUSSION TITLE
What is the status of the project?

### DISCUSSION THREAD

**Discussion by bigcat88** [heshengtao/comfyui_LLM_party#6] [General]
Good day.

I was faced with the problem that ComfyUI does not have normal nodes for LLM (so that both local and remote LLM and LVM would be supported) - apart from your project, I did not find anything useful.

I would like to thank you for the project and ask about your further plans for its development, will you develop it further?

I am specifically interested in working in all three major operating systems: Windows, macOS and Linux.
And work on Nvidia, AMD and Intel GPUs.

If you are going to further develop the project, can I ask you - are you ready to consider pull requests to correct bugs found/make installation easier?

**Comment by heshengtao** [comment]
Good day!

Thank you very much for your email and interest in the ComfyUI_LLM_party project. Please forgive me for replying with machine translation; if there are any stiff or misleading tones, please don't take it the wrong way, haha.

The positioning of my project is to provide a complete solution for building LLM workflows on comfyui. My next steps are planned in the following areas:
1. More model adaptations, at least covering the API interfaces of mainstream large models and local calls of mainstream open-source models, as well as more LVM model adaptations. Currently, I have only adapted the visual function calls of GPT-4;
2. More ways to build agents. The work I have completed in this area includes importing an LLM as a tool to another LLM, achieving radial construction of LLM workflows, and importing one workflow as a node into another workflow. I might develop some cooler functions in this area in the future.
3. More automation features. In the future, I will introduce more nodes that automatically push images, text, videos, and audio to other applications, as well as listening nodes that implement automatic replies to mainstream social software and forums.
4. More knowledge base management functions. The project already supports local file search and web search. In the future, I will introduce knowledge graph search and long-term memory search. This will allow agents to think logically about professional knowledge and always remember certain key information when conversing with users.
5. More tools, more persona. This part is the easiest to do but also requires the most accumulation. I hope that in the future, this project can have as many custom nodes as comfyui, with a multitude of tools and persona.

Regarding the cross-platform compatibility and different GPU compatibility issues you mentioned, I know very little about this area and greatly need the help of experts like you.

Regarding the pull requests you mentioned, I warmly welcome community contributions, whether it's fixing bugs or simplifying the installation process. Please feel free to submit your pull requests, and I will review and merge appropriate changes as soon as possible.

Thank you again for your support, and I look forward to further collaboration with you.

**Reply by bigcat88** [reply]
Thank you very much for your complete and detailed answer.

I will create issues with questions and suggestions then.

Then let's try these wonderful young nodes :)
